controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -29,21 +42,155 @@ type RegisteredClusterReconciler struct {
 	APIExtensionClient apiextensionsclient.Interface
 	Log                logr.Logger
 	Scheme             *runtime.Scheme
+	MceCluster         []helpers.MceInstance
 }
 
 func (r *RegisteredClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = context.Background()
 	logger := r.Log.WithValues("namespace", req.Namespace, "name", req.Name)
 	logger.Info("Reconciling...")
 
+	instance := &singaporev1alpha1.RegisteredCluster{}
+
+	if err := r.Client.Get(
+		context.TODO(),
+		types.NamespacedName{Namespace: req.Namespace, Name: req.Name},
+		instance,
+	); err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, giterrors.WithStack(err)
+	}
+
+	// create managecluster on creation of registeredcluster CR
+	if err := r.createManagedCluster(req.Name, ctx); err != nil {
+		logger.Error(err, "failed to create ManagedCluster")
+
+		return ctrl.Result{}, err
+	}
+
+	// update status of registeredcluster - add import command
+	if err := r.updateImportCommand(instance, req.Name, req.Namespace, ctx); err != nil {
+		logger.Error(err, "failed to update import command")
+
+		return ctrl.Result{}, err
+	}
+
 	return ctrl.Result{}, nil
 }
 
+func (r *RegisteredClusterReconciler) updateImportCommand(regCluster *singaporev1alpha1.RegisteredCluster, name string, namespace string, ctx context.Context) error {
+	// get import secret from mce managecluster namespace
+
+	importSecret := &corev1.Secret{}
+	if err := r.MceCluster[0].APIReader.Get(ctx, types.NamespacedName{Namespace: name, Name: name + "-import"}, importSecret); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+		return giterrors.WithStack(err)
+	}
+
+	applierBuilder := &clusteradmapply.ApplierBuilder{}
+	applier := applierBuilder.WithClient(r.KubeClient, r.APIExtensionClient, r.DynamicClient).Build()
+	readerDeploy := deploy.GetScenarioResourcesReader()
+
+	files := []string{
+		"cluster-registration-operator/import_configmap.yaml",
+	}
+
+	// Get yaml representation of import command
+	crdsYaml, err := yaml.Marshal(importSecret.Data["crds.yaml"])
+	crdsv1Yaml, err := yaml.Marshal(importSecret.Data["crdsv1.yaml"])
+
+	crdsv1beta1Yaml, err := yaml.Marshal(importSecret.Data["crdsv1beta1.yaml"])
+
+	importYaml, err := yaml.Marshal(importSecret.Data["import.yaml"])
+
+	values := struct {
+		Name        string
+		Namespace   string
+		CrdsYaml    string
+		CrdsV1Yaml  string
+		CrdsV1beta1 string
+		ImportYaml  string
+	}{
+		Name:        name,
+		Namespace:   namespace,
+		CrdsYaml:    string(crdsYaml),
```

Comment:
Do you know why there's all the different CRD versions in here? I was hoping we could prepare the whole import command for the user, same as our console code does today. Thoughts?
https://github.com/stolostron/console-ui/blob/b7f480e03c0a369b16ba843edd182df78c01a18f/src-web/components/importcluster/ImportClusterPage.js#L273

Reply:
Different CRD version according to kube version on managedcluster. It should be v1beta1 if managedcluster kube < 1.16 and v1 if managedcluster kube version > 1.16.
@itdove correct me if I'm wrong.

Reply:
you are correct, @leena-jawale